### DISCUSSION TITLE
Steep onboarding

### DISCUSSION THREAD

**Discussion by alper** [basecamp/kamal#1094] [General]
Trying to get this to work and… it's not very smooth.

## Registry

The docs don't tell you Docker hub also needs a user and there's no key generated for it in the secrets file.

So something like this:
```
registry:
  username:
    - KAMAL_REGISTRY_USER
  password:
    - KAMAL_REGISTRY_PASSWORD
```

## SSH

Don't really see how to get public key login to work. Logging in with a password is weird.

This does not do it:

```
ssh:
  keys_only: true
  keys: ["~/.ssh/id_ed25519"]
```

## Setup

Then we can finally do some setup, and it breaks on the fact that there's no git repository on the remote host?

```
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
  Finished all in 8.1 seconds
  ERROR (RuntimeError): Exception while executing on host 157.90.20.242: Can't use commit hash as version, no git repository found in /workdir
```

No idea but I'd have thought it would rsync the files over there for me.

**Comment by buncis** [comment]
as per key andd ssh config

you done it at .ssh/config 

kamal would load that automatically

also could u add more info about the code that you provided?

where is the file location/ what is the filename


do your project have git version control?

there is no rsync, kamal would just pull/download your package.

**Comment by alper** [comment]
It may be because of this:

```
servers:
  web:
    - 157.90.20.242
 ```

In my ssh_config this has a name but putting the name here wouldn't pick it up.

> there is no rsync, kamal would just pull/download your package

Then I still don't understand why it says there's no repository, but anyway, I think the tool is not for my use case.

**Comment by tuladhar** [comment]
Maybe checkout this beginner's to help you get started:
- kamal.purutuladhar.com

But I agree, we definitely need to improve the documentation for new users. Thanks for the feedback!
